Patch to make svcadm enable default to using the -r flag.  This patch
has not yet been submitted upstream.

--- puppet-3.8.6/lib/puppet/provider/service/smf.rb.orig        2016-04-19 14:53:31.762390677 -0700
+++ puppet-3.8.6/lib/puppet/provider/service/smf.rb     2016-04-19 14:56:32.596751150 -0700
@@ -74,7 +74,7 @@
     when :maintenance
       [command(:adm), :clear, @resource[:name]]
     else
-      [command(:adm), :enable, "-s", @resource[:name]]
+      [command(:adm), :enable, "-rs", @resource[:name]]
     end
   end
